### PR TITLE
fix ssl contrastive loss

### PR DESCRIPTION
**Pull request by stevehuang52**
Fix SSL contrastive loss when audio shorter than num negatives

### CODE REVIEW

File: nemo/collections/asr/losses/ssl_losses/contrastive.py [hunk 1]
```diff
@@ -137,7 +137,13 @@ def sample_negatives(self, y, num):
         # y - T'xBxC or T'xC
 
         high = y.shape[0]
-        neg_idxs = torch.multinomial(torch.ones((num, high), device=y.device), self.num_negatives)
+        # When fewer candidate frames are available than the number of negatives
+        # requested (e.g. very short utterances), sampling without replacement is
+        # impossible, so fall back to sampling with replacement.
```
Comment (nithinraok) [inline nemo/collections/asr/losses/ssl_losses/contrastive.py:142]: can we reduce length of comments.

**Review by nithinraok — APPROVED** [review]

**Comment by stevehuang52** [comment]
/ok to test [6f2d1c9](https://github.com/NVIDIA-NeMo/Speech/pull/15872/commits/6f2d1c9b0fb295e8db610d58abcae2befd4122cf)
